tests/test_annotator.py

```diff
@@ -18,6 +18,16 @@ class TestAnnotator(unittest.TestCase):
     def setUp(self):
         self.data = [[1, 2, 3], [2, 5, 7]]
         self.data2 = pd.DataFrame([[1, 2], [2, 5], [3, 7]], columns=["X", "Y"])
+        self.data3: dict[str, list[str]] = {
+            "x_axis": [1.01, 1.02, 1.03, 1.01, 1.02, 1.03, 1.01, 1.02, 1.03, 1.01, 1.02, 1.03, 1.01, 1.02, 1.03, 1.01,
+                       1.02, 1.03, 1.01, 1.02, 1.03, 1.01, 1.02, 1.03, 1.01, 1.02, 1.03, 1.01, 1.02, 1.03, 1.01, 1.02,
+                       1.03, 1.01, 1.02, 1.03, 1.01, 1.02, 1.03, 1.01, 1.02, 1.03, 1.01, 1.02, 1.03, 1.01, 1.02, 1.03,
+                       1.01, 1.02],
+            "y_axis": [16.99, 10.34, 21.01, 23.68, 24.59, 25.29, 8.77, 26.88, 15.04, 14.78, 10.27, 35.26, 15.42, 18.43,
+                       14.83, 21.58, 10.33, 16.29, 16.97, 20.65, 17.92, 20.29, 15.77, 39.42, 19.82, 17.81, 13.37, 12.69,
+                       21.7, 19.65, 9.55, 18.35, 15.06, 20.69, 17.78, 24.06, 16.31, 16.93, 18.69, 31.27, 16.04, 17.46,
+                       13.94, 9.68, 30.4, 18.29, 22.23, 32.4, 28.55, 18.04]
```

Comment:
What do you think about adding a third dimension and use it as hue (so with hue in pairs)?
If it works here it will also apply to integers which I did not yet cover either.
(Also, could you limit the data to maybe 3 lines reaching a max of 80 characters, so for the first line of `y_axis`, up to `26.88`.)

Reply:
Changes were committed.
However, I might be wrong, but I'm not sure using hue works. Can you tell me if I made a mistake with this minimal reproducible example?
```py
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt
from statannotations.Annotator import Annotator

x_axis = [
    1.01,
    1.02,
    1.03,
    1.01,
    1.02,
    1.03,
    1.01,
    1.02,
    1.03,
    1.01,
    1.02,
    1.03,
    1.01,
    1.02,
    1.03,
    1.01,
    1.02,
    1.03,
    1.01,
    1.02,
    1.03,
    1.01,
    1.02,
    1.03,
    1.01,
    1.02,
    1.03,
]
y_axis = [
    16.99,
    10.34,
    21.01,
    23.68,
    24.59,
    25.29,
    8.77,
    26.88,
    15.04,
    14.78,
    10.27,
    35.26,
    15.42,
    18.43,
    14.83,
    21.58,
    10.33,
    16.29,
    16.97,
    20.65,
    17.92,
    20.29,
    15.77,
    39.42,
    14.44,
    25.69,
    69.25,
]
hue = [
    "hue_2",
    "hue_2",
    "hue_1",
    "hue_1",
    "hue_2",
    "hue_2",
    "hue_2",
    "hue_1",
    "hue_2",
    "hue_1",
    "hue_1",
    "hue_2",
    "hue_1",
    "hue_1",
    "hue_0",
    "hue_0",
    "hue_2",
    "hue_1",
    "hue_0",
    "hue_0",
    "hue_2",
    "hue_0",
    "hue_2",
    "hue_1",
    "hue_0",
    "hue_0",
    "hue_2",
]

df = pd.DataFrame(data={"y_axis": y_axis, "x_axis": x_axis, "hue": hue})
x = "x_axis"
y = "y_axis"
hue = "hue"
hue_order = ["hue_0", "hue_1", "hue_2"]
order = [1.01, 1.02, 1.03]
pairs = [
    ((1.01, "hue_0"), (1.02, "hue_1"))
]

ax = sns.boxplot(data=df, x=x, y=y, order=order, hue=hue, hue_order=hue_order)

annotator = Annotator(ax, pairs, data=df, x=x, y=y, order=order)
annotator.configure(test="Mann-Whitney", text_format="star", loc="outside")
annotator.apply_and_annotate()
plt.savefig("test.png", dpi=300, bbox_inches="tight")
```

I get the following error:
```py
ValueError: Missing x value(s) `"(1.01, 'hue_0')", "(1.02, 'hue_1')"` in x_axis (specified in `pairs`) in data
```

I feel like I've done what was done in the section `Boxplot with hue` of the example notebook.

Reply:
Using
```py
pairs = [
    (("1.01", "hue_0"), ("1.01", "hue_1"))
]
```
doesn't work either.

Reply:
Changes were commmitted.